biz.aQute.resolve/src/biz/aQute/resolve/obr/plugin/ObrExporter.java

```diff
@@ -0,0 +1,85 @@
+package biz.aQute.resolve.obr.plugin;
+
+import static aQute.bnd.osgi.Constants.RESOLVE_EXCLUDESYSTEM;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.util.AbstractMap.SimpleEntry;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.osgi.resource.Resource;
+
+import aQute.bnd.annotation.plugin.BndPlugin;
+import aQute.bnd.build.Project;
+import aQute.bnd.osgi.FileResource;
+import aQute.bnd.osgi.repository.XMLResourceGenerator;
+import aQute.bnd.service.export.Exporter;
+import aQute.lib.exceptions.Exceptions;
+import biz.aQute.resolve.Bndrun;
+import biz.aQute.resolve.RunResolution;
+
+@BndPlugin(name = "OBR Exporter")
+public final class ObrExporter implements Exporter {
+
+    static final String TYPE = "exporter.obr";
+
+    @Override
+    public String[] getTypes() {
+        return new String[] { TYPE };
+    }
+
+    @Override
+    public Entry<String, aQute.bnd.osgi.Resource> export(
+            final String type,
+            final Project project,
+            final Map<String, String> options) throws Exception {
+
+        final String outputdir       = options.get("outputdir");
+        final String name            = options.getOrDefault("name", removeExtension(project.getName()) + ".xml");
+        final String excludesystem   = options.getOrDefault("excludesystem", "true");
+
+        File outputDirectory = null;
+        if (outputdir == null) {
+            outputDirectory = project.getTargetDir();
+        } else {
+            outputDirectory = new File(project.getBase(), outputdir);
+        }
+
+        project.setProperty(RESOLVE_EXCLUDESYSTEM, excludesystem);
+
+        final File obrIndex             =  new File(outputDirectory, name);
+        final RunResolution resolution  =  ((Bndrun) project).resolve();
+        final List<Resource> resources  =  resolution.getOrderedResources();
+
+        exportOBR(project, resources, obrIndex);
+
+        if (project.isOk()) {
+            final FileResource result = new FileResource(obrIndex);
+            return new SimpleEntry<>("obr", result);
+        }
+        return null;
+    }
+
+    private static void exportOBR(
+            final Project project,
+            final Collection<Resource> resources,
+            final File output) {
+
+        final XMLResourceGenerator xmlResourceGenerator = new XMLResourceGenerator();
+
+        try (final ByteArrayOutputStream buffer = new ByteArrayOutputStream()) {
+            xmlResourceGenerator.resources(resources);
+            xmlResourceGenerator.save(output);
+        } catch (final Exception e) {
+            project.error("Failed to generate OBR from %s due to %s", project, Exceptions.toString(e));
```

Comment:
I think you want to use the exception method here instead of error.

Reply:
Didn't know about it. I have seen one instance where `error` was used instead of `exception` and that's why I used the same.

Reply:
DONE ✔️